frontend/src/metabase/query_builder/components/filters/pickers/RelativeDatePicker.jsx

```diff
@@ -66,7 +66,7 @@ export default class RelativeDatePicker extends Component {
             typeof intervals === "number" ? Math.abs(intervals) : intervals
           }
           onChange={value =>
-            onFilterChange([op, field, formatter(value), unit])
+            onFilterChange([op, field, formatter(value), unit, options])
```

Comment:
This is going to add a `null` to the end of the filter even if it doesn't have options.

Ideally we'd expand the `Filter` wrapper class in `metabase-lib` to have various subclasses like we do for `Dimension` (which should probably be called `FieldReference`, FWIW) so you could just do `filter.setPeriod()` or `filter.setUnit()` etc which would carry over options if present. This kind of logic shouldn't be in React components to begin with.

I went ahead and pushed a kind of hacky solution just using `assoc` for now.